packages/itwin/grouping-mapping-widget/src/widget/components/SelectIModel.tsx

```diff
@@ -2,57 +2,133 @@
 * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
 * See LICENSE.md in the project root for license terms and full copyright notice.
 *--------------------------------------------------------------------------------------------*/
-import type { AccessToken } from "@itwin/core-bentley";
-import type {
-  ApiOverrides,
-  IModelFull,
-} from "@itwin/imodel-browser-react";
-import {
-  IModelGrid,
-} from "@itwin/imodel-browser-react";
-import { Button } from "@itwin/itwinui-react";
-import React, { useEffect, useState } from "react";
-import { useGroupingMappingApiConfig } from "./context/GroupingApiConfigContext";
+import type { EntityListIterator, IModel, IModelsClient } from "@itwin/imodels-client-management";
+import type { TablePaginatorRendererProps } from "@itwin/itwinui-react";
+import { Button, Table, tableFilters, TablePaginator } from "@itwin/itwinui-react";
+import React, { useCallback, useEffect, useMemo, useState } from "react";
+import type { CreateTypeFromInterface } from "../utils";
 import "./SelectIModel.scss";
+import type { GetAccessTokenFn } from "./context/GroupingApiConfigContext";
+import { useGroupingMappingApiConfig } from "./context/GroupingApiConfigContext";
+import { useIModelsClient } from "./context/IModelsClientContext";
+import { handleError } from "./utils";
+
+type IIModelTyped = CreateTypeFromInterface<IModel>;
+
+const defaultDisplayStrings = {
+  iModels: "iModels",
+  iModelName: "Name",
+  iModelDescription: "Description",
+};
+
+const fetchIModels = async (
+  setIModels: React.Dispatch<React.SetStateAction<IModel[]>>,
+  setIsLoading: React.Dispatch<React.SetStateAction<boolean>>,
```

Comment:
```suggestion
  setIModels: (iModels: IModel[]) => void,
  setIsLoading: (isLoading: boolean) => void,
```

Looks cleaner and is a bit more restrictive.

Reply:
Done